src/engine/strat_engine/setup.rs

```diff
@@ -26,6 +26,64 @@ use super::range_alloc::RangeAllocator;
 use super::serde_structs::PoolSave;
 
 
+/// Determine if devnode is a stratis device, if it is we will add  pool uuid and device
+/// information to pool_map and return pool uuid.
+pub fn is_stratis_device(pool_map: &mut HashMap<PoolUuid, HashMap<Device, PathBuf>>,
```

Comment:
I would be much happier if this did not take ```pool_map``` as an argument. That does mean that you have to insert into a map in two places, but that is better than having a significant side-effect buried in the method.